src/test/java/ay2122s1_cs2103t_w16_2/btbb/model/order/OrderTest.java

```diff
@@ -26,15 +29,21 @@ public void isSameOrder() {
         // null -> returns false
         assertFalse(ORDER_FOR_BENSON.equals(null));
 
-        // different phone -> returns false
-        Order editedRandomOrder = new OrderBuilder(ORDER_FOR_CARL).withClientPhone(new Phone(VALID_PHONE_BOB)).build();
+        Order editedRandomOrder = new OrderBuilder(ORDER_FOR_CARL).withClientName(new Name(VALID_NAME_BOB))
+                .withClientPhone(new Phone(VALID_PHONE_BOB))
+                .withClientAddress(new Address(VALID_ADDRESS_BOB)).build();
+        // different object, same client name, same client phone and same client address -> returns true
+        assertTrue(ORDER_FOR_BOB.equals(editedRandomOrder));
+
+        // different client phone -> returns false
+        editedRandomOrder = new OrderBuilder(ORDER_FOR_CARL).withClientPhone(new Phone(VALID_PHONE_BOB)).build();
         assertFalse(ORDER_FOR_CARL.equals(editedRandomOrder));
 
-        // different name -> returns false
+        // different client name -> returns false
         editedRandomOrder = new OrderBuilder(ORDER_FOR_DANIEL).withClientName(new Name(VALID_NAME_AMY)).build();
         assertFalse(ORDER_FOR_DANIEL.equals(editedRandomOrder));
```

Comment:
You are testing the wrong method for the entire `isSameOrder` test. You should be using the `isSameOrder` method and not the `equals` method.